test/framework/machinedeployment_helpers.go

```diff
@@ -137,3 +141,137 @@ func DiscoveryAndWaitForMachineDeployments(ctx context.Context, input DiscoveryA
 	}
 	return machineDeployments
 }
+
+// UpgradeMachineDeploymentAndWaitInput is the input type for UpgradeMachineDeploymentAndWait.
+type UpgradeMachineDeploymentAndWaitInput struct {
+	ClusterProxy                ClusterProxy
+	Cluster                     *clusterv1.Cluster
+	UpgradeVersion              string
+	WaitForMachinesToBeUpgraded []interface{}
+	MachineDeployments          []*clusterv1.MachineDeployment
+}
+
+// UpgradeMachineDeploymentAndWait upgrades a machine deployment and waits for its machines to be upgraded.
+func UpgradeMachineDeploymentAndWait(ctx context.Context, input UpgradeMachineDeploymentAndWaitInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for UpgradeMachineDeploymentAndWait")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling UpgradeMachineDeploymentAndWait")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling UpgradeMachineDeploymentAndWait")
+	Expect(input.UpgradeVersion).ToNot(BeNil(), "Invalid argument. input.KubernetesUpgradeVersion can't be nil when calling UpgradeMachineDeploymentAndWait")
+	Expect(input.MachineDeployments).ToNot(BeEmpty(), "Invalid argument. input.MachineDeployments can't be empty when calling UpgradeMachineDeploymentAndWait")
+
+	mgmtClient := input.ClusterProxy.GetClient()
+
+	for _, deployment := range input.MachineDeployments {
+		fmt.Fprintf(GinkgoWriter, "Patching the new kubernetes version to Machine Deployment\n")
+		patchHelper, err := patch.NewHelper(deployment, mgmtClient)
+		Expect(err).ToNot(HaveOccurred())
+
+		deployment.Spec.Template.Spec.Version = &input.UpgradeVersion
+		Expect(patchHelper.Patch(context.TODO(), deployment)).To(Succeed())
+
+		fmt.Fprintf(GinkgoWriter, "Waiting for machines to have the upgraded kubernetes version\n")
```

Comment:
should we add the namespace/name of the machine deployment being patched & the source/target Kubernetes version?